homeassistant/components/waze_travel_time/sensor.py

```diff
@@ -67,6 +87,39 @@ async def async_setup_entry(
     sensor = WazeTravelTime(config_entry.entry_id, name, origin, destination, data)
 
     async_add_entities([sensor], False)
+    if not hass.services.has_service(DOMAIN, SERVICE_UPDATE_TRAVEL_TIME):
+        async_setup_services(hass)
+
+
+@callback
+def async_setup_services(hass: HomeAssistant) -> None:
```

Comment:
Glad to see another service with return values!

I have some broader questions that come to mind.  Not that we need to handle all of this in this PR, but can we talk a little bit about what future additional features you had  in mind? For example, would we ever want to support service calls to return arbitrary routes between arbitrary origins/destinations in the future? If so then i could imagine a different service call (not an "update") but i'm not sure what the API would look like -- e.g. config entry as input instead of an entity.  Anyway, just curious if you had thoughts about the future direction

I generally don't want to increase the scope here too much, but this is really making me want to keep thinking about better/common APIs across some of the routing and travel time integrations ... e.g. a common device class or some service that can work across Waze, Maps, and Here. (but also on the input side, defining device classes for input location values)

Reply:
I see separate threads here:

1. We do see so many common elements that a `travel_time` platform makes sense now. I can provide standard properties like `duration`, `distance` and standard services like `update_travel_time` and `get_travel_time` where the latter could provide information on arbitrary origin/destination arguments like you mentioned. 
2. This brings up the question where this service should live. It shouldn't need a configured route between A->B just so I have the service available while it certainly does need a valid config entry so it can use credentials for `here_travel_time`and `google_travel_time`.
3. The next point is that we can introduce a `LocationEntity`. It is a entity which can be used by travel_time integrations as a origin/destination. For now `zone`, `device_tracker`, `person` have coordinate attributes but also `sensor` entities can hold coordinates as a state. Follow up questions would be if those `LocationEntities` should appear on maps.